### PR TITLE
Add CI to build aptos CLI for Windows

### PR DESCRIPTION
**Pull request by banool**
### Description
Originally I was trying to build on ubuntu-latest with the windows toolchain to avoid writing another script for building the CLI, but I realized that's silly, this is simpler and more native to Windows.

This is only possible now because of https://github.com/aptos-labs/aptos-core/pull/3751.

I confirmed that the built CLI works on Windows 11. I haven't been able to test `run-local-testnet` because I haven't been able to get my hands on an x86_64 machine (this particular command doesn't work under ARM windows, it just exits). Greg tested this as working earlier though, so we should be good there (and it's not relevant to this particular PR).

### Test Plan
On a Windows machine:
```
# Including since I'm a powershell noob, this enables script execution:
Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser

# Execute the script
.\scripts\cli\build_cli_release.ps1
```

I tested the new workflow job from my fork of aptos-core: https://github.com/banool/aptos-core/runs/8163538455?check_suite_focus=true. I confirmed that it's using the correct toolchain: 1.63.0-x86_64-pc-windows-msvc.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3785)

### CODE REVIEW

**Review by rustielin — APPROVED** [review]

**Review by jjleng — APPROVED** [review]
LGTM. Good news for Windows users

File: .github/workflows/cli-release.yaml [hunk 1]
```diff
@@ -50,11 +50,27 @@ jobs:
...
+      - name: Upload Binary
+        uses: actions/upload-artifact@v3
+        with:
+          name: cli-builds
+          path: aptos-cli-*.zip
+
   release-binaries:
     name: "Release binaries"
     needs:
       - build-linux-binary
```
Comment (gregnazario) [inline .github/workflows/cli-release.yaml:71]: btw later we should see why 20.04 builds don't work on 22.04 but that's fine

**Review by gregnazario — APPROVED** [review]

File: .github/workflows/cli-release.yaml [hunk 2]
```diff
@@ -22,7 +22,7 @@ on:
 jobs:
   build-linux-binary:
     name: "Build Linux binary"
```
Comment (perryjrandall) [inline .github/workflows/cli-release.yaml:24]: why did you remove those quotes but not these quotes XD

**Review by perryjrandall — COMMENTED** [review]
The horror...

jk its amazing
